### PR TITLE
Improve documentation for obsoletion/analyzer diagnostics

### PR DESCRIPTION
**Pull request by jeffhandley**
We've had some obsoletions go into effect recently where it became clear our documentation around handling this was insufficient. This PR augments the obsoletion doc so that we have better guidance available.

Note that this PR still leaves `SYSLIB0014` as absent from the list of obsoletions; PR #51550 adds that row to the table.

### CODE REVIEW

**Comment by jeffhandley** [comment]
@JeremyKuhne I tagged you for review on this since I'd like to make sure you think this documentation would have sufficiently helped you with your recent obsoletion. Please let me know if you have any suggestions to make it even better.

File: docs/project/list-of-diagnostics.md [hunk 1]
```diff
@@ -1,15 +1,44 @@
...
+    * Example: `[System.ObsoleteAttribute("The UTF-7 encoding is insecure and should not be used. Consider using UTF-8 instead.", DiagnosticId = "SYSLIB0001", UrlFormat = "https://aka.ms/dotnet-warnings/{0}")]`
+* If the library builds against downlevel targets earlier than .NET 5.0, then add an internal copy of `ObsoleteAttribute`
+    * The compiler recognizes internal implementations of `ObsoleteAttribute` to enable the `DiagnosticId` and `UrlFormat` properties to light up downlevel
+    * An MSBuild property can be added to the project's first `<PropertyGroup>` to achieve this easily
+    * Example: `<IncludeInternalObsoleteAttribute>true</IncludeInternalObsoleteAttribute>`
+    * This will need to be specified in both the `src` and `ref` projects
+* Apply the `breaking-change` label to the PR that introduces the obsoletion
+    * A bot will automatically apply the `needs-breaking-change-doc-created` label when the `breaking-change` label is detected
+* Follow up with the normal breaking change process to communicate and document the breaking change
+
```
Comment (danmoseley) [inline docs/project/list-of-diagnostics.md:32]: Does the aka.ms/dotnet-warnings/xxxx link automatically do something useful after the above steps, or is there another step for that?

Comment (jeffhandley) [reply docs/project/list-of-diagnostics.md:32]: Oh, good point, @danmoseley. I'll get those instructions added in.

File: docs/project/list-of-diagnostics.md [hunk 2]
```diff
@@ -1,15 +1,52 @@
...
+## Obsoletions
 
-When obsoleting an API, use the diagnostic ID `SYSLIB####`, where _\#\#\#\#_ is the next four-digit identifier in the sequence, and add it to the list below. This helps us maintain a centralized location of all APIs that were obsoleted using this mechanism.
+Per https://github.com/dotnet/designs/blob/master/accepted/2020/better-obsoletion/better-obsoletion.md, we now have a strategy for marking existing APIs as `[Obsolete]`. This takes advantage of the new diagnostic id and URL template mechanisms introduced to `ObsoleteAttribute` in .NET 5.
 
-The URL template we use for obsoletions is `https://aka.ms/dotnet-warnings/{0}`.
+The diagnostic id values reserved for obsoletions are `SYSLIB0001` through `SYSLIB0999`. When obsoleting an API, claim the next three-digit identifier in the `SYSLIB0###` sequence and add it to the list below. The URL template for all obsoletions is `https://aka.ms/dotnet-warnings/{0}`. The `{0}` placeholder is replaced by the compiler with the `SYSLIB0###` identifier.
```
Comment (stephentoub) [inline docs/project/list-of-diagnostics.md:7]: > values reserved for obsoletions are `SYSLIB0001` through `SYSLIB0999`

Can you add a comment to https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/System/Obsoletions.cs about the expected range?

**Review by stephentoub — APPROVED** [review]

File: docs/project/list-of-diagnostics.md (hunk 2 above)
Comment (jeffhandley) [reply docs/project/list-of-diagnostics.md:7]: Good idea. I added a comment block in that file both pointing to this doc and showing the expected range.

**Comment by jeffhandley** [comment]
@JeremyKuhne I went ahead and merged but please let me know if you have any feedback/suggestions and I can augment if needed.

**Comment by JeremyKuhne** [comment]
It looks much better @jeffhandley, thanks!

**Pull request merged**